CMakeLists.txt

```diff
@@ -1,60 +1,44 @@
-project(PROGRESSBAR_FOR_A_MORE_CIVILIZED_AGE)
-cmake_minimum_required(VERSION 2.8.5)
+project(elegant-progressbar-for-a-more-civilized-age-dev)
+cmake_minimum_required(VERSION 3.0.2)
 
 
 ###############################################################################
 # Using Clang is awesome
 # code based on http://stackoverflow.com/q/7031126
 ###############################################################################
-option(USE_CLANG "build application with clang" ON) # OFF is the default
+string(FIND ${CMAKE_CXX_COMPILER} "clang++" CXX_COMPILER_IS_CLANG++)
+string(FIND ${CMAKE_CXX_COMPILER} "g++" CXX_COMPILER_IS_G++)
+
 option(CMAKE_BUILD_TYPE "build type" Slow) # to show the progress
-if(USE_CLANG)
-  SET (CMAKE_C_COMPILER             "/usr/bin/clang")
-  SET (CMAKE_C_FLAGS                "-Wall -Wextra -std=c99")
-  SET (CMAKE_C_FLAGS_DEBUG          "-g")
-  SET (CMAKE_C_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
-  SET (CMAKE_C_FLAGS_RELEASE        "-O3 -DNDEBUG")
-  SET (CMAKE_C_FLAGS_RELWITHDEBINFO "-O2 -g")
-  SET (CMAKE_C_FLAGS_SLOW           "-O0")
 
-  SET (CMAKE_CXX_COMPILER             "/usr/bin/clang++")
+if(${CXX_COMPILER_IS_CLANG++})  
   SET (CMAKE_CXX_FLAGS                "-Wall -Wextra -std=c++11")
   SET (CMAKE_CXX_FLAGS_DEBUG          "-g")
   SET (CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
   SET (CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
   SET (CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
   SET (CMAKE_CXX_FLAGS_SLOW           "-O0")
-else(USE_CLANG)
-  SET (CMAKE_C_COMPILER             "/usr/bin/gcc")
-  SET (CMAKE_C_FLAGS                "-Wall -Wextra -std=c99")
-  SET (CMAKE_C_FLAGS_DEBUG          "-g")
-  SET (CMAKE_C_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
-  SET (CMAKE_C_FLAGS_RELEASE        "-O3 -DNDEBUG")
-  SET (CMAKE_C_FLAGS_RELWITHDEBINFO "-O2 -g")
-  SET (CMAKE_C_FLAGS_SLOW           "-O0")
-
-  SET (CMAKE_CXX_COMPILER             "/usr/bin/g++")
-  SET (CMAKE_CXX_FLAGS                "-Wall -Wextra -std=c++11")
+endif(${CXX_COMPILER_IS_CLANG++})
+  
+if(${CXX_COMPILER_IS_G++})  
+  SET (CMAKE_CXX_FLAGS                "-Wall -Wextra -std=c++11")  
   SET (CMAKE_CXX_FLAGS_DEBUG          "-g")
   SET (CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
   SET (CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
   SET (CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
   SET (CMAKE_CXX_FLAGS_SLOW           "-O0")
-endif(USE_CLANG)
-
-
+endif(${CXX_COMPILER_IS_G++})  
 
 ###############################################################################
-# Headers
+# Progressbar for a More Civilized Age Config
 ###############################################################################
 # for header-only libraries, this is enough :)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
-
+find_package(elegant-progressbar-for-a-more-civilized-age 1.0.0 REQUIRED PATHS ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
The cmake config-files should be located in a subdirectory (I think)... that requires adaptions in the PATHS argument